be/src/storage/column_expr_predicate.cpp

```diff
@@ -224,16 +238,14 @@ Status ColumnExprPredicate::try_to_rewrite_for_zone_map_filter(starrocks::Object
     }
     // build new ColumnExprPredicates
     for (auto& expr : exprs_after_rewrite) {
-        ExprContext* ctx = pool->add(new ExprContext(expr));
-        RETURN_IF_ERROR(ctx->prepare(_state));
-        RETURN_IF_ERROR(ctx->open(_state));
-        pool->add(new DeferOp([ctx, this]() { ctx->close(_state); }));
+        auto expr_rewrite = std::make_unique<ExprContext>(expr);
+        RETURN_IF_ERROR(expr_rewrite->prepare(_state));
+        RETURN_IF_ERROR(expr_rewrite->open(_state));
+
         ColumnExprPredicate* new_pred =
-                pool->add(new ColumnExprPredicate(_type_info, _column_id, _state, ctx, _slot_desc));
-        // copy other cast exprs
-        for (size_t i = 1; i < _expr_ctxs.size(); i++) {
-            new_pred->_add_expr_ctx(_expr_ctxs[i]);
-        }
+                pool->add(new ColumnExprPredicate(_type_info, _column_id, _state, nullptr, _slot_desc));
+        new_pred->_add_expr_ctx(std::move(expr_rewrite));
+        new_pred->_add_expr_ctxs(_expr_ctxs);
```

Comment:
_exprs_ctxs is '=' '>' '<' we only need add '>=' and '<='

Reply:
fixed